src/windows/enumerate.rs

```diff
@@ -401,36 +401,36 @@ fn get_registry_com_ports() -> HashSet<String> {
         };
         if SUCCEEDED(query_res) {
             for idx in 0..num_key_values {
-                let mut val_name_buff = [0i8; MAX_PATH];
+                let mut val_name_buff = [0u16; MAX_PATH];
                 let mut val_name_size = MAX_PATH as u32;
                 let mut value_type = 0;
                 // if 100 chars is not enough for COM<number> something is very wrong
-                let mut val_data = [0; 100];
-                let mut data_size = val_data.len() as u32;
+                let mut val_data = [0u16; 100];
+                let mut byte_len = 2 * val_data.len() as u32; // len doubled
```

Comment:
Should't these be checked multiplications because we easily can?

Reply:
I'm unsure as to the motivation for this? There doesn't appear to be any other uses of the checked operations in the lib and the default behaviour is to assert in Debug/test profiles if an overflow is encountered

Reply:
My motivation in the first place was to catch overflows. But they won't happen with this code as we are using fixedly sized buffers. So you are right, this is nothing we need to take care of in the short term.